pwiz_tools/Skyline/TestUtil/ScreenshotManager.cs

```diff
@@ -0,0 +1,309 @@
+                throw new NotImplementedException();
+            }
+
+            public override void SetUp()
+            {
+            }
+
+            public override Bitmap Take(Form activeWindow)
+            {
+                throw new NotImplementedException();
+            }
+
+        }
+
+        [DllImport("user32.dll")]
+        public static extern bool SetProcessDPIAware();
+
+        private string filePath
+        {
+            get
+            {
+                var exeDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+                return Path.Combine(exeDir ?? "", _tutorialName + "_shots");
+            }
+        }
+
+        public ScreenshotManager([NotNull] string pTutorialName, [NotNull] SkylineWindow pSkylineWindow)
+        {
+            //look up the settings file, read and parse if found
+            //set up defaults otherwise
+            _tutorialName = pTutorialName;
+
+
+
+            _storage = new XmlDocument();
+
+            if (File.Exists(filePath))
+            {
+                _storage.Load(filePath);
+                XmlNode root = _storage.DocumentElement;
+                if (root.HasChildNodes)
+                {
+                    foreach (XmlNode shotNode in root.ChildNodes)
+                    {
+                        _shotSequence.Add(SkylineScreenshot.createScreenshot(pSkylineWindow, shotNode));
+                    }
+                }
+            }
+            else
+                _storage.AppendChild(_storage.CreateElement(ROOT_ELEMENT));
+
+            _currentShotIndex = -1;
+            SetProcessDPIAware();
+        }
+
+
+        public void TakeNextShot(Form activeWindow)
+        {
+            _skylineWindow = Program.MainWindow;
+            if (activeWindow == null)
+                activeWindow = _skylineWindow;
+            Bitmap shotPic;
+            if ( ++_currentShotIndex < _shotSequence.Count)
+            {
+                shotPic = _shotSequence[_currentShotIndex].Take(activeWindow);
+            }
+            else
+            {
+                //check UI and create a blank shot according to the user selection
+                SkylineScreenshot newShot = new ActiveWindowShot(_skylineWindow);
+                _shotSequence.Add(newShot);
+                _currentShotIndex = _shotSequence.Count - 1;
+                shotPic = _shotSequence.Last().Take(activeWindow);
+                _storage.DocumentElement.AppendChild(newShot.Serialize(_storage));
+                SaveToFile();
+            }
+
+            if (shotPic != null)
+            {
+                //Have to do it this way because of the limitation on OLE access from background threads.
+                Thread clipThread = new Thread(() => Clipboard.SetImage(shotPic));
+                clipThread.SetApartmentState(ApartmentState.STA);
+                clipThread.Start();
+                clipThread.Join();
+            }
+        }
+
+        public void SaveToFile()
```

Comment:
I think this method should take the file path as a parameter, and the caller should be the one deciding where the screenshot should be saved (and it would have something to do with TestContext.TestDir).

Reply:
Well, TestContext.TestDir points to the "SkylineTester Results" folder that is volatile and dropped/recreated every time the test is run. 
The idea here is for the user to create a customized set of screenshot configurations that can be re-used on subsequent runs, so the location should be persistent. Also, since the file is saved after each change there is no need to call it externally. I actually removed the call to SaveToFile() from the AbstractFunctionalTest as redundant and made the method private.